### PR TITLE
Removed bias from Conv2D if followed by BatchNorm.

### LINKED ISSUE #629: Redundant bias paramter in ResNet's conv2d.
**Short description**
In `tf.keras` by default Conv2D has bias enabled. We should explicitly set `use_bias=False` if the convolution layer is followed by BatchNormalization.

This is the done in some existing Keras-CV implementations:
* DarkNet Conv Block - [link](https://github.com/keras-team/keras-cv/blob/f3e32b5d74d69419c5ec17b04f9ca96807474ed1/keras_cv/models/__internal__/darknet_utils.py#L28)
* DenseNet - [link](https://github.com/keras-team/keras-cv/blob/f3e32b5d74d69419c5ec17b04f9ca96807474ed1/keras_cv/models/densenet.py#L134)
* ResNet V2 - [link](https://github.com/keras-team/keras-cv/blob/f3e32b5d74d69419c5ec17b04f9ca96807474ed1/keras_cv/models/resnet_v2.py#L116)

But not in `resnet_v1` - all ResNet's have [bias enabled](https://github.com/keras-team/keras-cv/blob/f3e32b5d74d69419c5ec17b04f9ca96807474ed1/keras_cv/models/resnet_v1.py#L113), which causes redundant computation and is not aligned with other implementations.

I think this is rather a quick fix - add `use_bias=False` parameter to the Conv2d's in the `Block` function. If you approve I could submit a PR.

### PR DESCRIPTION
**Pull request by sebastian-sz**
# What does this PR do?
Remove bias parameter from ResNetV1 Conv2d if the next layer is batch normalization.
Batch Normalization eliminates the need for bias, as it is in Densenet or ResNetV2. 

Fixes #629 

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/keras-team/keras-cv/blob/master/.github/CONTRIBUTING.md),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue? Please add a link
      to it if that's the case.
- [ ] Did you write any new necessary tests?
- [ ] If this adds a new model, can you run a few training steps on TPU in Colab to ensure that no XLA incompatible OP are used?

## Who can review?
@LukeWood @ianstenbit

### CODE REVIEW

**Comment by LukeWood** [comment]
So is this handled by the BatchNorm layers?  Do they have a bias built in?

**Comment by sebastian-sz** [comment]
@LukeWood yes below is a summary from Nvidia's [performance tuning guide](https://nvlabs.github.io/eccv2020-mixed-precision-tutorial/files/szymon_migacz-pytorch-performance-tuning-guide.pdf) that eventually landed in [Pytorch docs](https://pytorch.org/tutorials/recipes/recipes/tuning_guide.html#disable-bias-for-convolutions-directly-followed-by-a-batch-norm)

>Disable bias for convolutions directly followed by a batch norm
>
>If a nn.Conv2d layer is directly followed by a nn.BatchNorm2d layer, then the bias in the convolution is not needed, instead use nn.Conv2d(..., bias=False, ....). Bias is not needed because in the first step BatchNorm subtracts the mean, which effectively cancels out the effect of bias.

**Comment by LukeWood** [comment]
> @LukeWood yes below is a summary from Nvidia's [performance tuning guide](https://nvlabs.github.io/eccv2020-mixed-precision-tutorial/files/szymon_migacz-pytorch-performance-tuning-guide.pdf) that eventually landed in [Pytorch docs](https://pytorch.org/tutorials/recipes/recipes/tuning_guide.html#disable-bias-for-convolutions-directly-followed-by-a-batch-norm)
> 
> > Disable bias for convolutions directly followed by a batch norm
> > If a nn.Conv2d layer is directly followed by a nn.BatchNorm2d layer, then the bias in the convolution is not needed, instead use nn.Conv2d(..., bias=False, ....). Bias is not needed because in the first step BatchNorm subtracts the mean, which effectively cancels out the effect of bias.

@fchollet and compatibility concerns here?  Seems like the move to keras_cv.models is a good time to do this

**Review by LukeWood — APPROVED** [review]

**Comment by LukeWood** [comment]
Thank you @sebastian-sz
